tidb: add the refresh stats statement (#21873)
pingcap/docs: sql-statements/sql-statement-refresh-stats.md

```ebnf+diagram
RefreshStatsStmt ::=
    'REFRESH' 'STATS' RefreshTargetList RefreshMode? ClusterOption?

RefreshTargetList ::=
    RefreshTarget (',' RefreshTarget)*

RefreshTarget ::=
    TableName
  | SchemaWildcard
  | GlobalWildcard

TableName ::=
    Identifier ('.' Identifier)?

SchemaWildcard ::=
    Identifier '.' '*'

GlobalWildcard ::=
    '*' '.' '*'

RefreshMode ::=
    'FULL'
  | 'LITE'

ClusterOption ::=
    'CLUSTER'
```

## Options

- **Targets (`RefreshTargetList`)**
    - `table_name` refreshes a table in the current database.
    - `db_name.table_name` refreshes a fully qualified table.
    - `db_name.*` refreshes every table in the specified database.
    - `*.*` refreshes every table in the cluster.
- **`FULL`** Forces TiDB to load full statistics (histograms, TopN, CMSketches) into memory, equivalent to setting [`lite-init-stats`](/tidb-configuration-file.md#lite-init-stats-new-in-v710) to `false` for this operation. Use this when you need complete statistics immediately after the refresh.
- **`LITE`** Uses lightweight initialization, equivalent to `lite-init-stats = true`, which skips loading histograms and other heavy structures until they are needed.

> - **`CLUSTER`** Broadcasts the refresh request to every TiDB server. Without this keyword, only the TiDB node that receives the statement reloads its in-memory statistics.

Reviewer: ```suggestion
- **`FULL`**: loads complete statistics (such as histograms, TopN, and CMSketches) into memory, equivalent to setting [`lite-init-stats`](/tidb-configuration-file.md#lite-init-stats-new-in-v710) to `false` for this operation. Use this when you need complete statistics immediately.
- **`LITE`**: performs lightweight initialization, equivalent to `lite-init-stats = true`, which skips loading histograms and other heavy structures until they are needed.
- **`CLUSTER`**: broadcasts the refresh request to all TiDB servers. Without this option, only the TiDB node that receives the statement reloads its in-memory statistics.
```

> **Warning:**
>
> This statement is designed for the internal restore workflow of BR, and BR runs it automatically when needed. 
> 
> - **DO NOT** execute this statement manually unless you are debugging in-memory statistics, and never run it on production clusters as part of routine operations.
> - **DO NOT** execute this statement concurrently on the same tables, as it might lead to unexpected behavior.

## Synopsis

```ebnf+diagram
RefreshStatsStmt ::=
    'REFRESH' 'STATS' RefreshTargetList RefreshMode? ClusterOption?

RefreshTargetList ::=
    RefreshTarget (',' RefreshTarget)*

RefreshTarget ::=
    TableName
  | SchemaWildcard
  | GlobalWildcard

TableName ::=
    Identifier ('.' Identifier)?

SchemaWildcard ::=
    Identifier '.' '*'

GlobalWildcard ::=
    '*' '.' '*'

RefreshMode ::=
    'FULL'
  | 'LITE'

ClusterOption ::=
    'CLUSTER'
```

## Options

- **Targets (`RefreshTargetList`)**:
    - `table_name` refreshes a table in the current database.

>     - `db_name.table_name` refreshes a fully qualified table.

Reviewer: ```suggestion
    - `db_name.table_name` refreshes a table in the specified database.
```

    - `db_name.*` refreshes every table in the specified database.

>     - `*.*` refreshes every table in the cluster.

Reviewer 2: ```suggestion
    - `table_name`: refreshes a table in the current database.
    - `db_name.table_name`: refreshes a table in the specified database.
    - `db_name.*`: refreshes every table in the specified database.
    - `*.*`: refreshes every table in the cluster.
```

> - **`FULL`**: loads complete statistics (such as histograms, TopN, and CMSketches) into memory, equivalent to setting [`lite-init-stats`](/tidb-configuration-file.md#lite-init-stats-new-in-v710) to `false` for this operation. Use this when you need complete statistics immediately.

Reviewer 2: ```suggestion
- **`FULL`**: loads complete statistics (such as histograms, Top-N, and Count-Min sketches) into memory, equivalent to setting [`lite-init-stats`](/tidb-configuration-file.md#lite-init-stats-new-in-v710) to `false` for this operation. Use this option when you need complete statistics immediately.
```

- **`LITE`**: performs lightweight initialization, equivalent to `lite-init-stats = true`, which skips loading histograms and other heavy structures until they are needed.
- **`CLUSTER`**: broadcasts the refresh request to every TiDB server. Without this option, only the TiDB node that receives the statement reloads its in-memory statistics.

> - **Default mode**: If neither `FULL` nor `LITE` is specified, TiDB uses the current value of [`lite-init-stats`](/tidb-configuration-file.md#lite-init-stats-new-in-v710).

Reviewer 2: ```suggestion
- **Default mode**: if neither `FULL` nor `LITE` is specified, TiDB uses the value of [`lite-init-stats`](/tidb-configuration-file.md#lite-init-stats-new-in-v710).
```

## Examples

- Refresh statistics for a single table on the connected TiDB node:

  ```sql
  REFRESH STATS orders;
  ```

- Refresh all tables in `sales` with lightweight initialization:

  ```sql
  REFRESH STATS sales.* LITE;
  ```

- Force every TiDB node to load complete statistics for the entire cluster:

  ```sql
  REFRESH STATS *.* FULL CLUSTER;
  ```

## Privileges

> To execute `REFRESH STATS`, you must either have the `RESTORE_ADMIN` privilege or have the `SELECT` privilege on every target table. If your privileges are insufficient, TiDB returns an error and aborts the statement execution.

Reviewer 2: ```suggestion
To execute `REFRESH STATS`, you must have either the `RESTORE_ADMIN` privilege or the `SELECT` privilege on every target table. If your privileges are insufficient, TiDB returns an error and aborts the statement.
```

## MySQL compatibility

> `REFRESH STATS` is a TiDB-specific extension and is not part of MySQL.

Reviewer 2: ```suggestion
`REFRESH STATS` is a TiDB extension to MySQL syntax.
```

    - `db_name.*` refreshes every table in the specified database.
    - `*.*` refreshes every table in the cluster.
- **`FULL`** Forces TiDB to load full statistics (histograms, TopN, CMSketches) into memory, equivalent to setting [`lite-init-stats`](/tidb-configuration-file.md#lite-init-stats-new-in-v710) to `false` for this operation. Use this when you need complete statistics immediately after the refresh.
- **`LITE`** Uses lightweight initialization, equivalent to `lite-init-stats = true`, which skips loading histograms and other heavy structures until they are needed.
- **`CLUSTER`** Broadcasts the refresh request to every TiDB server. Without this keyword, only the TiDB node that receives the statement reloads its in-memory statistics.
- **Default mode** If neither `FULL` nor `LITE` is specified, TiDB uses the current value of [`lite-init-stats`](/tidb-configuration-file.md#lite-init-stats-new-in-v710).

## Examples

- Refresh statistics for a single table on the connected TiDB node:

  {{< copyable "sql" >}}

  ```sql
  REFRESH STATS orders;
  ```

- Refresh all tables in `sales` with lightweight initialization:

  {{< copyable "sql" >}}

> 

Reviewer: ```suggestion
```
